### DISCUSSION TITLE
[Proposal]: Introduce static variable declarations within methods

### DISCUSSION THREAD

**Discussion by taori** [dotnet/csharplang#3907] [General]
Sometimes i write code which declares objects in a fashion where access to them will never be required by other members and their contents will never change.

In order to optimize my code up until now, i usually make use of static readonly field declarations. Ideally however i would like to do this in the scope of a member, instead of on a class level, to reduce the namespace polution, since if there are multiple methods declaring their resources as static readonly i will have many distractions in intellisense from the actual code i want to write.

That way my class would turn from

```cs
public class RetryScheduler {
    private static readonly int[] DelayIterations = new [] { 1000, 30000, 60000 };
    public void Execute(Action action) {
        foreach( var delay in DelayIterations) {
            await Task.Delay(delay);
            action?.Invoke();
        }
    }
}
```

to this

```cs
public class RetryScheduler {
    public void Execute(Action action) {
        static var delays = new [] { 1000, 30000, 60000 };
        foreach( var delay in delays) {
            await Task.Delay(delay);
            action?.Invoke();
        }
    }
}
```

**Comment by svick** [comment]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/832.
